# HG changeset patch
# User rk
# Date 1285859449 -7200
# Node ID 42b7d73a172065758d3f7698e7142ac7fa1a31aa
# Parent  9a535123d884b23ebe38968c6b346f3575d615cb
fixed the test in wxAuiManager::DoDrop() that determines if dropping a window to an edge of the frame (see wxWidgets trac ticket #10612)

diff -r 9a535123d884 -r 42b7d73a1720 src/aui/framemanager.cpp
--- a/src/aui/framemanager.cpp	Thu Sep 30 16:30:46 2010 +0200
+++ b/src/aui/framemanager.cpp	Thu Sep 30 17:10:49 2010 +0200
@@ -2888,7 +2888,9 @@ bool wxAuiManager::DoDrop(wxAuiDockInfoA
 
 
     if (pt.x < layer_insert_offset &&
-        pt.x > layer_insert_offset-auiLayerInsertPixels)
+        pt.x > layer_insert_offset-auiLayerInsertPixels &&
+        pt.y > 0 &&
+        pt.y < cli_size.y)
     {
         int new_layer = wxMax(wxMax(GetMaxLayer(docks, wxAUI_DOCK_LEFT),
                                 GetMaxLayer(docks, wxAUI_DOCK_BOTTOM)),
@@ -2904,7 +2906,9 @@ bool wxAuiManager::DoDrop(wxAuiDockInfoA
         return ProcessDockResult(target, drop);
     }
     else if (pt.y < layer_insert_offset &&
-             pt.y > layer_insert_offset-auiLayerInsertPixels)
+             pt.y > layer_insert_offset-auiLayerInsertPixels &&
+             pt.x > 0 &&
+             pt.x < cli_size.x)
     {
         int new_layer = wxMax(wxMax(GetMaxLayer(docks, wxAUI_DOCK_TOP),
                                 GetMaxLayer(docks, wxAUI_DOCK_LEFT)),
@@ -2920,7 +2924,9 @@ bool wxAuiManager::DoDrop(wxAuiDockInfoA
         return ProcessDockResult(target, drop);
     }
     else if (pt.x >= cli_size.x - layer_insert_offset &&
-             pt.x < cli_size.x - layer_insert_offset + auiLayerInsertPixels)
+             pt.x < cli_size.x - layer_insert_offset + auiLayerInsertPixels &&
+             pt.y > 0 &&
+             pt.y < cli_size.y)
     {
         int new_layer = wxMax(wxMax(GetMaxLayer(docks, wxAUI_DOCK_RIGHT),
                                 GetMaxLayer(docks, wxAUI_DOCK_TOP)),
@@ -2936,7 +2942,9 @@ bool wxAuiManager::DoDrop(wxAuiDockInfoA
         return ProcessDockResult(target, drop);
     }
     else if (pt.y >= cli_size.y - layer_insert_offset &&
-             pt.y < cli_size.y - layer_insert_offset + auiLayerInsertPixels)
+             pt.y < cli_size.y - layer_insert_offset + auiLayerInsertPixels &&
+             pt.x > 0 &&
+             pt.x < cli_size.x)
     {
         int new_layer = wxMax( wxMax( GetMaxLayer(docks, wxAUI_DOCK_BOTTOM),
                                       GetMaxLayer(docks, wxAUI_DOCK_LEFT)),
